prober/http.go

```diff
@@ -403,6 +406,10 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 		request.Header.Set(key, value)
 	}
 
+	if request.Header.Get("User-Agent") == "" {
```

Comment:
Hmm, what if the user explicitly wants no User-Agent header?

Reply:
Initially I was looking for Has method to check if key exists but it is not exported.
What do you think about this solution ?
(Since all headers are added with Set() we are sure that the key was formatted to canonical key)

```
	_, hasUserAgent := request.Header["User-Agent"]
	if !hasUserAgent {
		request.Header.Set("User-Agent", userAgentDefaultHeader)
	}
```
blackbox.yaml
```
headers:
  // Result : User-Agent: "My user agent"
  User-Agent: "My user agent"

headers:
  // Result : User-Agent: ""
  User-Agent: ""

headers:
  // Result : User-Agent: "Blackbox Exporter/0.16"
  User-Agent:

headers:
  // Result : User-Agent: "Blackbox Exporter/0.16"
  User-Agent: null
```

Reply:
I think it'll do for now, we can tune the null case in future if someone asks.